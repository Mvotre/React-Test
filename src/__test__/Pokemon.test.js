import React from "react";
import { render, cleanup } from "@testing-library/react";
import '@testing-library/jest-dom';
import Pokemon from '../components/pokemon';

afterEach(cleanup);

it("matches snapshot", () => {
  const { asFragment } = render( <Pokemon />);
  expect(asFragment()).toMatchSnapshot();
})


it("pass proper name", () => {
  const { getByTestId } = render( <Pokemon nome={"Zémon"} />);
  const nomePok = getByTestId('nome');
  expect(nomePok.textContent).toBe("Nome: Zémon")
})
