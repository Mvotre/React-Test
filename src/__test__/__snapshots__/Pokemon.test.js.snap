// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="pk-singleBox"
  >
    <img
      alt="Pokemon"
    />
    <p
      data-testid="nome"
    >
      <span>
        Nome:
      </span>
       
    </p>
    <p>
       
      <span>
         ID: 
      </span>
        
    </p>
    <p>
       
      <span>
         Tipo: 
      </span>
        
    </p>
  </div>
</DocumentFragment>
`;
